cmd/core-command/res/configuration.toml

```diff
@@ -51,7 +51,11 @@ Protocol = 'https'
 RootCaCertPath = '/vault/config/pki/EdgeXFoundryCA/EdgeXFoundryCA.pem'
 ServerName = 'localhost'
 TokenFile = '/vault/config/assets/resp-init.json'
-  [SecretStore.Authentication]
+# Number of attemtps to retry retrieving secrets before failing to start the service.
+AdditionalRetryAttempts = 10
+# Amount of time to wait before attempting another retry in nanoseconds.
+RetryWaitPeriod =  1000000000
```

Comment:
I think this is wrong, we shouldn't be using nanoseconds (i.e. ints) here for the time, we should be using a string that is parsed into a time.Duration.

See how I did this for docker-edgex-mongo here: https://github.com/edgexfoundry/docker-edgex-mongo/pull/59

Reply:
(What Ian said.)

Reply:
I'll go forward with this suggestion. However, I disagree with how this is being achieved. In the `docker-edgex-mongo` PR we are duplicating the `SecretConfig` struct and the one that lives in `docker-edgex-mongo` has diverged from `go-mod-secrets` and the `RetryWaitPeriod` is a `string`:

https://github.com/edgexfoundry/docker-edgex-mongo/blob/4c0243a28f29f77b6a324602919251fb9d36520f/internal/pkg/config.go#L64

where the version that exists in `go-mod-secrets` it is a `time.Duration`:

https://github.com/edgexfoundry/go-mod-secrets/blob/f1caf4c75e0add8e50e323efbf869f084199d470/pkg/providers/vault/config.go#L34

My vision for the core-services, is to be ignorant to the underlying configuration for the `SecretClient`(as much as possible), so that we can swap out different versions of `go-mod-secrets` and the only thing we would have to do is change configuration. Keeping on that philosophy, I think it would be better if either:

1. `go-mod-secrets:SecretConfig` is updated so that `RetryWaitPeriod` is an `int` and we pick a default unit of time.

2. Update the `go-mod-secrets` to perform the `time.Duration` parsing and we can simply pass the configuration to `go-mod-secrets and this will be consistent for all consumers.

Just my thoughts

Reply:
I'll go forward with this suggestion. However, I disagree with how this is being achieved. In the `docker-edgex-mongo` PR we are duplicating the `SecretConfig` struct and the one that lives in `docker-edgex-mongo` has diverged from `go-mod-secrets` and the `RetryWaitPeriod` is a `string`:

https://github.com/edgexfoundry/docker-edgex-mongo/blob/4c0243a28f29f77b6a324602919251fb9d36520f/internal/pkg/config.go#L64

where the version that exists in `go-mod-secrets` it is a `time.Duration`:

https://github.com/edgexfoundry/go-mod-secrets/blob/f1caf4c75e0add8e50e323efbf869f084199d470/pkg/providers/vault/config.go#L34

My vision for the core-services, is to be ignorant to the underlying configuration for the `SecretClient`(as much as possible), so that we can swap out different versions of `go-mod-secrets` and the only thing we would have to do is change configuration. Keeping on that philosophy, I think it would be better if either:

1. `go-mod-secrets:SecretConfig` is updated so that `RetryWaitPeriod` is an `int` and we pick a default unit of time.

2. Update the `go-mod-secrets` to perform the `time.Duration` parsing and we can simply pass the configuration to `go-mod-secrets and this will be consistent for all consumers.

Just my thoughts

Reply:
@AnthonyMBonafide 

> so that we can swap out different versions of go-mod-secrets and the only thing we would have to do is change configuration

what's the use case for this? It kinda sounds like you're saying "don't break APIs" in go-mod-secrets (i.e. the different versions of go-mod-secrets being just future versions).

I'm +1 on not breaking APIs, but having slightly inconsistent configuration.toml derived structs and SecretConfig structs passed to go-mod-secrets to me is okay, because really the problem we have is that go-mod-secrets is something that talks "go" (and thus gets cool and correct typing with time.Duration) and these other things like docker-edgex-mongo are things that talk "user provided config files" (and thus only get basic strings and numbers). That's a more general problem though and I'd rather push that complexity to clients like docker-edgex-mongo than have that complexity in go-mod-secrets. But that's a personal preference...

FWIW if you feel strongly about this I'd rather go with 2 and have the parsing happen in go-mod-secrets than have all configuration.tomls provide the wait time in nanoseconds.